contrib/check-config.sh

```diff
@@ -163,7 +163,7 @@ printf -- '- '
 if [ "$(stat -f -c %t /sys/fs/cgroup 2> /dev/null)" = '63677270' ]; then
 	wrap_good 'cgroup hierarchy' 'cgroupv2'
 else
-	cgroupSubsystemDir="$(awk '/[, ](cpu|cpuacct|cpuset|devices|freezer|memory)[, ]/ && $3 == "cgroup" { print $2 }' /proc/mounts | head -n1)"
+	cgroupSubsystemDir="$(grep -Em1 '^[^ ]+ [^ ]+ cgroup ([^ ]*,)?(cpu|cpuacct|cpuset|devices|freezer|memory)[, ]' /proc/mounts | while read -r _ dir _; do echo "$dir"; done)"
```

Comment:
If we're going to go to this level, IMO we might as well switch to `sed` and do it in a single command:

```suggestion
	cgroupSubsystemDir="$(sed -rne '/^[^ ]+ ([^ ]+) cgroup ([^ ]*,)?(cpu|cpuacct|cpuset|devices|freezer|memory)[, ].*$/ { s//\1/p; q }' /proc/mounts)"
```

(Having a shell looping construct in here at all feels like overkill.)

Reply:
So the loop is to give the positive result even for empty inputs, IMO not overkill but just standard shell scripting.

But I do like your `sed` version the best, that one is indeed the best, hands down. Outstanding winner.
With that I guess it's a bit wrong that the PR is tied to me now that you have the best solution. Can we find a good way to point the cred of the PR to you in the git history?

Reply:
...perhaps you could create a new PR with this final `sed` fix and then I close this PR and refer to that new PR that is yours?

Reply:
I don't need any credit, but if you want to make sure I get some you can add `Co-authored-by: Tianon Gravi <admwiggin@gmail.com>` to your commit message (especially since you're going to want/need to squash this all back down to one commit :sweat_smile:) -- your latest commit already has the appropriate `Co-authored-by` so you can see that GitHub's given us both credit on it. :smile:

All your commits have a valid DCO and you've enabled repository maintainers editing your branch, so I'd be happy to perform that squash for you if you'd like me to.

Reply:
> All your commits have a valid DCO and you've enabled repository maintainers editing your branch, so I'd be happy to perform that squash for you if you'd like me to.

No worries, git squash and push done now, and the commit message updated accordingly.